test/Kestrel.Core.Tests/Http2ConnectionTests.cs

```diff
@@ -2061,6 +2061,17 @@ public Task HEADERS_Received_HeaderBlockContainsTEHeader_ValueIsNotTrailers_Stre
 
             await SendGoAwayAsync();
 
+            var spinLimit = 50; // 2.5 seconds
```

Comment:
It might be slightly better/faster to hook to a state change event. In some other Kestrel tests, we just set a TCS when we see a specific log indicating a state transition. A test-only callback might also be acceptable.

Reply:
State changes currently are not logged and I don't think there's a good way of adding a test-only callback that doesn't at least add a tiny overhead. I'll see what it looks like.